src/meds_torch/models/components/ecg_encoder.py

```diff
@@ -85,15 +87,15 @@ def __init__(self, cfg):
 
         # Sinusoidal positional encodings for time dimension.
         self.time_pos_enc = self._generate_sinusoidal_positional_encoding(self.num_patches, self.token_dim)
-        self.register_buffer('time_pos_enc_buffer', self.time_pos_enc)
+        self.register_buffer("time_pos_enc_buffer", self.time_pos_enc)
 
         # Class token (learned parameter).
         self.cls_token = nn.Parameter(torch.zeros(1, 1, self.token_dim))
 
         # Transformer Encoder with 4 layers.
-        encoder_layer = nn.TransformerEncoderLayer(d_model=self.token_dim, nhead=8)
+        encoder_layer = nn.TransformerEncoderLayer(d_model=self.token_dim, nhead=2)
```

Comment:
We should consider making nheads a configurable parameter in a future commit. Had to reduce 8 to 2 so that it was divisible by test data token embedding dimension.